finality-aleph/src/sync/substrate/verification/cache.rs

```diff
@@ -128,9 +156,68 @@ where
     /// Prune all sessions with a number smaller than `session_id`
     fn prune(&mut self, session_id: SessionId) {
         self.sessions.retain(|&id, _| id >= session_id);
+        // off by one!
+        // let's retain one more session, just to be safe
```

Comment:
hmm I dont like this :P could you do the same with self.sessions and update function comment

Reply:
ok, the comment is "slightly" misleading :) I actually want to have one more session in `aura_authorities`, and the reason is sth like that: we set `CACHE_SIZE = 2`, so I'd expect that we can verify every header/justification from the current or previous session; since header verification requires access to its parent, we need to keep the additional past session to be able to verify the first header of the previous session - you still don't like it? if so, explain! :D if you do, I'll update the comment accordingly

Reply:
ooookay, this is buggy - let's do it your way

Reply:
so I've changed it back to the more reasonable behaviour (`CACHE_SIZE` defines cache size...), and changed the const from 2 to 3 everywhere
also, I've added a comment that this is necessary due to Aura, and slightly modified one test that assumed `CACHE_SIZE == 2`
please take a look at the commit to make sure that it looks fine now